src/txn/txn_timestamp.c

```diff
@@ -854,50 +854,67 @@ __wt_txn_set_read_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t read_ts)
-        set_ts = true;
+    /*
+     * If the API received no configuration string, there's nothing to do.
+     */
+    if (cfg == NULL || cfg[0] == NULL || cfg[1] == NULL)
+        return (0);
+
+    /*
+     * We take a shortcut in parsing that works because we're only given a base configuration and a
+     * user configuration.
+     */
+    WT_ASSERT(session, cfg[0] != NULL && cfg[1] != NULL && cfg[2] == NULL);
+    __wt_config_init(session, &cparser, cfg[1]);
+    while ((ret = __wt_config_next(&cparser, &ckey, &cval)) == 0) {
+        WT_ASSERT(session, ckey.str != NULL);
+        if (WT_STRING_MATCH("commit_timestamp", ckey.str, ckey.len)) {
+            WT_RET(__wt_txn_parse_timestamp(session, "commit", &commit_ts, &cval));
+            set_ts = true;
+        } else if (WT_STRING_MATCH("durable_timestamp", ckey.str, ckey.len)) {
+            WT_RET(__wt_txn_parse_timestamp(session, "durable", &durable_ts, &cval));
+            set_ts = true;
+        } else if (WT_STRING_MATCH("prepare_timestamp", ckey.str, ckey.len)) {
+            WT_RET(__wt_txn_parse_timestamp(session, "prepare", &prepare_ts, &cval));
+            set_ts = true;
+        } else if (WT_STRING_MATCH("read_timestamp", ckey.str, ckey.len)) {
+            WT_RET(__wt_txn_parse_timestamp(session, "durable", &read_ts, &cval));
+            set_ts = true;
+        }
     }
+    WT_RET_NOTFOUND_OK(ret);
+
+    /* Look for a commit timestamp. */
+    if (commit_ts != 0)
+        WT_RET(__wt_txn_set_commit_timestamp(session, commit_ts));
 
     /*
      * Look for a durable timestamp. Durable timestamp should be set only after setting the commit
      * timestamp.
      */
-    ret = __wt_config_gets_def(session, cfg, "durable_timestamp", 0, &cval);
-    WT_RET_NOTFOUND_OK(ret);
-    if (ret == 0 && cval.len != 0) {
-        WT_RET(__wt_txn_parse_timestamp(session, "durable", &ts, &cval));
-        WT_RET(__wt_txn_set_durable_timestamp(session, ts));
-    }
-
+    if (durable_ts != 0)
+        WT_RET(__wt_txn_set_durable_timestamp(session, durable_ts));
     __wt_txn_publish_durable_timestamp(session);
 
     /* Look for a read timestamp. */
-    WT_RET(__wt_config_gets_def(session, cfg, "read_timestamp", 0, &cval));
-    if (ret == 0 && cval.len != 0) {
-        WT_RET(__wt_txn_parse_timestamp(session, "read", &ts, &cval));
-        set_ts = true;
-        WT_RET(__wt_txn_set_read_timestamp(session, ts));
-    }
+    if (read_ts != 0)
+        WT_RET(__wt_txn_set_read_timestamp(session, read_ts));
 
     /* Look for a prepare timestamp. */
-    WT_RET(__wt_config_gets_def(session, cfg, "prepare_timestamp", 0, &cval));
-    if (ret == 0 && cval.len != 0) {
-        WT_RET(__wt_txn_parse_timestamp(session, "prepare", &ts, &cval));
-        WT_RET(__wt_txn_set_prepare_timestamp(session, ts));
-    }
+    if (prepare_ts != 0)
+        WT_RET(__wt_txn_set_prepare_timestamp(session, prepare_ts));
+
     if (set_ts)
         WT_RET(__wt_txn_ts_log(session));
```

Comment:
So related to the fix you did, can you update one or more of the `test_debug_mode` python tests to set/use durable and prepare timestamps and check the output.

Reply:
Added a test to check that this is working.